# Common configuration for the mavel velocity controller

# Control Rates
control_rate: 50.0

throttle:
  min: 0.2
  min: 0.5
  min: 0.9

# Maximum roll/pitch in radians
tilt_max: 0.39

# Data Stream
stream:
  min_rate:
    reference:
      position: 10.0
      velocity: 20.0
    setpoint:
      position: 5.0
      velocity: 10.0
      acceleration: 20.0
  topic:
    reference:
      position: "reference/position"
      velocity: "reference/velocity"
    setpoint:
      position: "setpoint/position"
      velocity: "setpoint/velocity"
      acceleration: "setpoint/acceleration"
    feedback:
      position: "goal/position"
      velocity: "goal/velocity"
      acceleration: "goal/acceleration"
      attitude: "goal/attitude"

# PID Controllers
pid:
  position:
    x:
      kp: 0.10
      ki: 0.0
      kd: 0.02
      tau: 0.05
      min: -1.0
      max: 1.0
    y:
      kp: 0.10
      ki: 0.0
      kd: 0.02
      tau: 0.05
      min: -1.0
      max: 1.0
    z:
      kp: 0.10
      ki: 0.0
      kd: 0.02
      tau: 0.05
      min: -0.5
      max: 0.5
  velocity:
    x:
      kp: 0.5
      ki: 0.0
      kd: 0.0
      tau: 0.05
      min: -0.2
      max: 0.2
    y:
      kp: 0.5
      ki: 0.0
      kd: 0.0
      tau: 0.05
      min: -0.2
      max: 0.2
    z:
      kp: 0.5
      ki: 0.0
      kd: 0.0
      tau: 0.05
      min: -0.2
      max: 0.2

